var/spack/repos/builtin/packages/cuda/package.py

```diff
@@ -691,6 +691,7 @@ class Cuda(Package):
 
     provides("opencl@:1.2", when="@7:")
     provides("opencl@:1.1", when="@:6")
+    provides("cub")
```

Comment:
Yeah - I wasn't sure how to add in version map between cuda and cub, and if there were cuda versions that didn't include cub. So it would be good if someone who knows this stuff can add in the corresponding changes.

As of now - mixing cub (latest) with cuda (any randomly chosen) version can trigger such errors - so needs a fix. An improved fix (over this PR) would be great!

Reply:
For reference: The packaged CUB version is documented in the CUDA Toolkit release notes: [most recent](https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html), [older (version in URL)](https://docs.nvidia.com/cuda/archive/12.6.2/cuda-toolkit-release-notes/index.html).

Reply:
Release notes for cuda versions older than 11.8.0 don't list cub versions [so this info is incomplete]

I filled in a few (from cuda installed include files) - perhaps you can fill more in - if you have any of these versions locally installed and can check.

(I guess we can added in guesses for the others - don't know how accurate this needs to be). well there some are discrepancies in these release notes vs:
```
./cuda-12.2.2-7djbt2eqrpadujvffssym4fyrm22jt76/include/cub/version.cuh:#define CUB_VERSION 200200
./cuda-12.4.1-ik4zuvjcne6ntypsukb3646fyazxqwzj/include/cub/version.cuh:#define CUB_VERSION 200302 // macro expansion with ## requires this to be a single value
```

Reply:
It is not ideal, but one can look at the tags in the [old CUB repo](https://github.com/NVIDIA/cub). There are tags both for CUB versions and for CUDA TK releases. From what I see the versions for CUDA 11.2-11.6 were always constant among patch releases, so you can just use the values you found for all patch releases corresponding ta each minor release.

Reply:
Updated. Thanks!

For now, will ignore versions older than 11.2